blocks/subscriptions-block/components/useCartOrderDetail.jsx

```diff
@@ -0,0 +1,300 @@
+							setIsFetchingCartOrder(false);
+						});
+					} else {
+						// There are no items into the cart, we will check last pending order & items in localStorage
+						let lastPendingOrder;
+						getLastPendingOrder()
+							.then((order) => {
+								if (order?.orderNumber) {
+									// There are pending order
+									getOrderDetail(order?.orderNumber)
+										.then((orderDetails) => {
+											if (orderDetails?.billingAddress?.country) {
+												lastPendingOrder = orderDetails;
+											}
+
+											if (lastPendingOrder?.orderNumber || itemCart?.cartDate) {
+												// getting the most recent one between pending order & cart
+												getLastestOrderOrCart(lastPendingOrder, itemCart).then(
+													({ lastOrder, currentCart }) => {
+														if (lastOrder?.items) {
+															getItemDetails(origin, lastOrder?.items).then((newDetails) => {
+																setOrderDetail({ ...lastOrder, items: newDetails });
+																setIsFetchingCartOrder(false);
+															});
+														}
+														if (currentCart?.items) {
+															getItemDetails(origin, currentCart?.items).then((newDetails) => {
+																setCartDetail({ ...currentCart, items: newDetails });
+																setIsFetchingCartOrder(false);
+															});
+														}
+													},
+												);
+											} else {
+												setIsFetchingCartOrder(false);
+											}
+										})
+										.catch(() => {
+											if (itemCart) {
+												createNewCart(itemCart)
+													.then((newCart) => {
+														getItemDetails(origin, newCart?.items).then((newDetails) => {
+															setCartDetail({ ...newCart, items: newDetails });
+															setIsFetchingCartOrder(false);
+														});
+													})
+													.catch(() => {
+														setIsFetchingCartOrder(false);
+													});
+											} else {
+												setIsFetchingCartOrder(false);
+											}
+										});
+								} else if (itemCart) {
+									// There are no pending order
+									getLastestOrderOrCart(undefined, itemCart).then(({ currentCart }) => {
+										getItemDetails(origin, currentCart?.items).then((newDetails) => {
+											setCartDetail({ ...currentCart, items: newDetails });
+											setIsFetchingCartOrder(false);
+										});
+									});
+								} else {
+									setIsFetchingCartOrder(false);
+								}
+							})
+							.catch(() => {
+								if (itemCart) {
+									createNewCart(itemCart)
+										.then((newCart) => {
+											getItemDetails(origin, newCart?.items).then((newDetails) => {
+												setCartDetail({ ...newCart, items: newDetails });
+												setIsFetchingCartOrder(false);
+											});
+										})
+										.catch(() => {
+											setIsFetchingCartOrder(false);
+										});
+								} else {
+									setIsFetchingCartOrder(false);
+								}
+							});
+					}
+				});
+			}
```

Comment:
All I have to say is wow... that's a set of promises...

Reply:
Yep @nschubach I know, but with this hook and all these promises we are trying to have all the information we need on Checkout block. 
The info coming from retail is rendered on the sidebar and Review section.
1st. In subs we have 3 verticals: identity, retail, sales. On the new checkout block we are providing info about retail (rates, price description, product description, product features). There is not an API able to return all this info. 
2nd. We don't have a mechanism to move a cart from anonymous user to logged In users. For this reason, we are trying to grab info from localStorage (info saved by an anonymous user), and create a cart.
3th. Also, we are trying to use pending orders. The current checkout block sometimes causes a lot of back and forth since, there we are just considering the current cart.
For the reasons mentioned above we have all these logic and promises there.